olive/cli/capture_onnx.py

```diff
@@ -0,0 +1,228 @@
+            ),
+        )
+        pte_group.add_argument(
+            "--target_opset",
+            type=int,
+            default=17,
+            help="The target opset version for the ONNX model. Default is 17.",
+        )
+
+        # Model Builder options
+        mb_group = sub_parser.add_argument_group("Model Builder options")
+        mb_group.add_argument(
+            "--use_model_builder",
+            action="store_true",
+            help="Whether to use Model Builder to capture ONNX model.",
+        )
+        mb_group.add_argument(
+            "--precision",
+            type=str,
+            default="fp16",
+            choices=["fp16", "fp32", "int4"],
+            help="The precision of the ONNX model. This is used by Model Builder",
+        )
+        mb_group.add_argument(
+            "--int4_block_size",
+            type=int,
+            required=False,
+            choices=[16, 32, 64, 128, 256],
+            help="Specify the block_size for int4 quantization. Acceptable values: 16/32/64/128/256.",
+        )
+        mb_group.add_argument(
+            "--int4_accuracy_level",
+            type=ModelBuilderAccuracyLevel,
+            required=False,
+            help="Specify the minimum accuracy level for activation of MatMul in int4 quantization.",
+        )
+        mb_group.add_argument(
+            "--exclude_embeds",
+            type=bool,
+            default=False,
+            required=False,
+            help="Remove embedding layer from your ONNX model.",
+        )
+        mb_group.add_argument(
+            "--exclude_lm_head",
+            type=bool,
+            default=False,
+            required=False,
+            help="Remove language modeling head from your ONNX model.",
+        )
+        mb_group.add_argument(
+            "--enable_cuda_graph",
+            type=bool,
+            default=None,  # Explicitly setting to None to differentiate between user intent and default.
+            required=False,
+            help=(
+                "The model can use CUDA graph capture for CUDA execution provider. "
+                "If enabled, all nodes being placed on the CUDA EP is the prerequisite "
+                "for the CUDA graph to be used correctly."
+            ),
+        )
+
+        # remote options
+        add_remote_options(sub_parser)
+
+        sub_parser.set_defaults(func=CaptureOnnxGraphCommand)
+
+    def run(self):
+        from olive.workflows import run as olive_run
+
+        set_tempdir(self.args.output_path)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            run_config = self.get_run_config(tempdir)
+
+            olive_run(run_config)
+
+            if is_remote_run(self.args):
+                # TODO(jambayk): point user to datastore with outputs or download outputs
+                # both are not implemented yet
+                return
+
+            output_path = Path(self.args.output_path)
```

Comment:
the output structure of olive runs are not intuitive. I think doing something like https://github.com/microsoft/Olive/blob/ce92e7997991fdffc8013e0987bbe4be88b0ab09/olive/cli/finetune.py#L164 might be better until we do some work on improving the output folder structure